### DISCUSSION TITLE
User analytics dashboard for mocks

### DISCUSSION THREAD

**Discussion by ave165** [microcks/microcks#1403] [Community]
Hi TEam,

Is there a way to extract data around who built what mock api, basically we want a feature where we want to know the owner of the mock api, and also the users who are using it.

Thanks
Abhinav

**Comment by yada** [comment]
Hi @ave165,

Thanks for reaching out! It’s great to hear you're using Microcks, and your question suggests that you might be utilizing it on a large scale.
If so, please consider joining our adopters list; this is important for the project to level up with the CNCF. 
See: https://github.com/microcks/.github/blob/main/ADOPTERS.md

Regarding your points:

**1/ Identifying Mock Owners:**

Assign the admin role exclusively to mock owners for easy identification. Use labels to tag them as "owners" (similar to how teams are organized in the documentation). For details, refer to the doc here: [Organizing Repository](https://microcks.io/documentation/guides/administration/organizing-repository/).

To retrieve a list of all owners, you can query labels via the Microcks API by using the /services/labels endpoint, which is outlined in the [API documentation](https://microcks.io/documentation/references/apis/open-api/). This list can then be used as needed.

Note that these labels are also visible in our Backstage plugin, which enhances integration. For more on this, check the Backstage integration blog post: [Backstage Integration Launch](https://microcks.io/blog/backstage-integration-launch/).

**2/ Tracking User Access:**

There’s no built-in authentication tracking for tracing mocks users generated by Microcks. For this purpose, an external authentication layer can be implemented before accessing the mock services, allowing you to monitor, report and control user access effectively.

Hope this helps! Please keep us updated.
